bundle/src/main/java/com/adobe/acs/commons/ondeploy/scripts/OnDeployScriptBase.java

```diff
@@ -0,0 +1,223 @@
+     *
+     * @param node Resource to update the property name on.
+     * @param oldPropertyName Old property name.
+     * @param newPropertyName New property name.
+     */
+    protected void renameProperty(Resource resource, String oldPropertyName, String newPropertyName) {
+        ModifiableValueMap properties = resource.adaptTo(ModifiableValueMap.class);
+        if (properties.containsKey(oldPropertyName)) {
+            logger.info("Renaming property '{}' to '{}' on resource: {}", new Object[] { oldPropertyName, newPropertyName, resource.getPath() });
+            properties.put(newPropertyName, properties.get(oldPropertyName));
+            properties.remove(oldPropertyName);
+        } else {
+            logger.debug("Property '{}' does not exist on resource: {}", oldPropertyName, resource.getPath());
+        }
+    }
+
+    /**
+     * Remove node at a given path.
+     *
+     * @param path Path to the node to remove.
+     */
+    protected void removeResource(String path) throws RepositoryException {
+        Resource resource = resourceResolver.getResource(path);
+        if (resource != null) {
+            logger.info("Deleting node at {}", path);
+            session.removeItem(path);
+        } else {
+            logger.info("Node at {} has already been removed", path);
+        }
+    }
+
+    /**
+     * Searches for the current sling:resourceType under /content and replaces any nodes it finds
+     * with the newResourceType.
+     *
+     * @param oldResourceType The current sling:resourceType.
+     * @param newResourceType The new sling:resourceType to be used.
+     */
+    protected void searchAndUpdateResourceType(String oldResourceType, String newResourceType) throws RepositoryException {
+        Map<String, String> map = new HashMap<>();
+        map.put("p.limit", "-1");
+        map.put("path", "/content");
+        map.put("1_property", SLING_RESOURCE_TYPE);
+        map.put("1_property.value", oldResourceType);
+
+        logger.info("Finding all nodes under /content with resource type: {}", oldResourceType);
+
+        Query query = queryBuilder.createQuery(PredicateGroup.create(map), session);
+        SearchResult result = query.getResult();
+        Iterator<Node> nodeItr = result.getNodes();
+        if (nodeItr.hasNext()) {
+            while (nodeItr.hasNext()) {
+                Node node = nodeItr.next();
+                updateResourceType(node, newResourceType);
+            }
+        } else {
+            logger.info("No nodes found with resource type: {}", oldResourceType);
+        }
+    }
+
+    /**
+     * Update the sling:resourceType of a node.
+     *
+     * @param node         The node to update.
+     * @param resourceType The new sling:resourceType to be used.
+     */
+    protected void updateResourceType(Node node, String resourceType) throws RepositoryException {
+        String currentResourceType = node.getProperty(SLING_RESOURCE_TYPE).getString();
```

Comment:
Should this do a hasProperty() check if someone else calls this (thats not searchAndUpdateResourceType()) ?

Reply:
yes that's a good call, it really should.